test/transfer_rule/constructor.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const web3 = require('../test_lib/web3.js');
+const utils = require('../test_lib/utils.js');
+
+const TokenRules = artifacts.require('TokenRules');
+
+const TransferRule = artifacts.require('TransferRule');
+const EIP20TokenMock = artifacts.require('EIP20TokenMock');
+const TokenRulesPassingGlobalConstraint = artifacts.require('TokenRulesPassingGlobalConstraint');
+
+contract('TransferRule::constructor', async () => {
+	contract('Negative testing for input parameters:', async (accounts) => {
+		
+		it('Token rules address is empty', async () => {
+			
+			const organization = accounts[0];
+			const constraint = await TokenRulesPassingGlobalConstraint.new();
+			const ruleName = 'A';
+			const entity1 = accounts[1];
+			const entity2 = accounts[4];
+			const ruleAbi = `Rule abi of ${ruleName}`;
+			const token = await EIP20TokenMock.new(
+				1, 1, 'OST', 'Open Simple Token', 1
+			);
+			
+			token.setBalance(entity1, 100);
+			
+			const tokenRulesInstance = await TokenRules.new(organization, token.address);
+			
+			await tokenRulesInstance.allowTransfers({from: entity1});
+			await tokenRulesInstance.addGlobalConstraint(constraint.address, {from: organization});
+			
+			await utils.expectRevert(TransferRule.new(''));
```

Comment:
Better validate for exact error message like below:

 await utils.expectRevert(
                TokenRules.new(''),
                'Should revert as tokenrule address is null.',
                'Token rules address is null.',
            );